src/main/java/graphql/schema/DataFetcher.java

```diff
@@ -31,4 +33,21 @@
     T get(DataFetchingEnvironment environment) throws Exception;
 
 
+    /**
+     * This is called to by the engine to get a value from the source object
+     *
+     * @param fieldDefinition     the graphql field definition
+     * @param sourceObject        the source object to get a value from
+     * @param environmentSupplier a supplier of the {@link DataFetchingEnvironment}
+     *
+     * @return a value of type T. May be wrapped in a {@link graphql.execution.DataFetcherResult}
+     *
+     * @throws Exception to relieve the implementations from having to wrap checked exceptions. Any exception thrown
+     *                   from a {@code DataFetcher} will eventually be handled by the registered {@link graphql.execution.DataFetcherExceptionHandler}
+     *                   and the related field will have a value of {@code null} in the result.
+     */
+    default T get(GraphQLFieldDefinition fieldDefinition, Object sourceObject, Supplier<DataFetchingEnvironment> environmentSupplier) throws Exception {
+        return get(environmentSupplier.get());
+    }
+
```

Comment:
old DataFetchers will be called via delegating back to the old method.  Those DFS that implement this method will be called direct